hw/ip/spi_host/doc/_index.md

```diff
@@ -1288,7 +1288,168 @@ Highlights for this unit:
+
+A few examples related to using SPI flash devices on a Little-Endian platform:
+- A 4-byte address can be loaded into the TX FIFO as four individual bytes using 8-bit I/O instructions.
+- The above read command (with 4-byte address) can be loaded into the FIFO by first loading the command code into {{< regref "DATA" >}} as a single byte, and the address can be loaded into {{< regref "DATA" >}} using 32-bit instructions, provided the byte order is swapped before loading.
+- Flash commands with 3-byte addressing require some care, as there are no 24-bit I/O instructions, though there are a several options:
+    - After the 8-bit command code is sent, the address can either be sent in several IO commands (e.g., the MSB is sent as an 8-bit command, and the remaining 16-bits can be sent after swapping)
+    - If bandwidth efficiency is a priority, the address, `A[23:0]`, and command code, `C[7:0]`, can all be packed together into a single 4-byte quantity `W[31:0] = {A[7:0], A[15:8], A[23:16], C[7:0]}`, which when loaded into {{< regref "DATA" >}} will ensure that the command code is sent first, followed by the address in MSB-first order.
+
+#### Segmenting Considerations
+
+Data words are *not* shared across segments.
+If at the end of each TX (or bidirectional) segment there is a partially transmitted data word then any unsent bytes will be discarded as the SPI_HOST IP closes the segment.
+For the next TX segment, the transmitted data will start with the following *word* from the TX FIFO.
+
+#### Refilling the TX FIFO
+
+For extremely long transactions, the TX FIFO may not have enough capacity to hold all the data being transmitted.
+In this case software can either poll the {{< regref "STATUS.TXQD" >}} register to determine the number of elements in the TX FIFO, or enable the SPI_HOST IP to send an interrupt when the FIFO drains to a certain level.
+If {{< regref "INTR_ENABLE.spi_event" >}} and {{< regref "EVENT_ENABLE.TXWM" >}} are both asserted, the IP will send an interrupt whenever the number of elements in the TX FIFO falls below {{< regref "CONTROL.TX_WATERMARK" >}}.
+
+### Specifying the Segments
+
+Each write to the {{< regref "COMMAND" >}} register corresponds to a single command segment.
+The length, CSAAT flag, direction and speed settings for that segment should all be packed into a single 32-bit register and written simultaneously to {{< regref "COMMAND" >}}.
+
+The {{< regref "COMMAND" >}} should only be written when {{< regref "STATUS.READY" >}} is asserted.
+
+While each command segment is being processed, the SPI_HOST has room to queue up exactly one additional segment descriptor in the Command CDC.
+Once a second command segment descriptor has been submitted, software must wait for the FSM to finish processing the current segment before submitting more.
```

Comment:
"wait for the FSM to finish" -> "wait for the state machine to finish" or "what for the SPI_HOST to finish".